app/models/result.rb

```diff
@@ -130,6 +131,9 @@ def self.get_total_extra_marks(result_ids, max_mark: nil, user_visibility: :ta_v
           assignment_max_mark = max_mark_hash[assessment_id]
         end
         extra_marks_hash[id] += (extra_mark * assignment_max_mark / 100).round(2)
+      elsif unit == 'percentage_of_score'
```

Comment:
Yikes, it's not great that we were using the hardcoded strings here. Please replace this (and the other two) with the constants from `ExtraMark`.